tests/jballerina-unit-test/src/test/resources/test-src/statements/variabledef/record-variable-definition-stmt.bal

```diff
@@ -774,6 +774,57 @@ function testNestedRecordVariableWithAnonymousType() {
     assertTrue(married);
 }
 
+type PersonWithContact record {|
+    string name;
+    record {|
+        "fax"|"mobile" 'type;
+        string number;
+    |}[2] contact;
+|};
+
+function testRecordTypedBindingPatternAgainstRecordLiteralInRecordDestructuring() {
+    record {|string state; int postalCode;|} {state, postalCode} = {state: "Alabama", postalCode: 35004};
+    assertEquality("Alabama", state);
+    assertEquality(35004, postalCode);
+
+    record {|string street; float...;|} {street, ...coordinates} = {street: "Highway 61", "lat": 37.30, "lon": -90.40};
```

Comment:
Also add for when some of the required fields are bound as rest.

Reply:
Added in https://github.com/ballerina-platform/ballerina-lang/pull/40283/commits/edebb78afebb8f6e60721465d50959820908d968